starknet_py/net/udc_deployer/deployer.py

```diff
@@ -64,20 +64,52 @@ async def prepare_contract_deployment(
         :param calldata: Constructor args of the contract to be deployed
         :param max_fee: Max amount of Wei to be paid when executing transaction
         :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
-        :return: InvokeFunction
+        :return: NamedTuple with invoke_tx and address fields
         """
+        call, address = await self.create_deployment_call(
+            class_hash=class_hash, salt=salt, abi=abi, calldata=calldata
+        )
+
+        transaction = await self.account.sign_invoke_transaction(
+            calls=call, max_fee=max_fee, auto_estimate=auto_estimate
+        )
+
+        return namedtuple(
```

Comment:
Let's use a NamedTuple from [typing](https://docs.python.org/3/library/typing.html#typing.NamedTuple) so we can have better typing. 

We can define the tuples outside methods and then use them for both type annotations and creating actual tuples.